receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -430,6 +431,28 @@ func parseMessageToMetric(line string, enableMetricType bool, enableSimpleTags b
 
 				kvs = append(kvs, attribute.String(k, v))
 			}
+		case strings.HasPrefix(part, "c:"):
+			// As per DogStatD protocol v1.2:
+			// https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics#dogstatsd-protocol-v12
+			containerID := strings.TrimPrefix(part, "c:")
+
+			if containerID != "" {
+				kvs = append(kvs, attribute.String("container_id", containerID))
```

Comment:
Should we use [OTel semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/4f434da570cda9c6ab7af3b5cfb797cfd2cbcd97/docs/attributes-registry/container.md?plain=1#L14) instead?

Reply:
That's a good point. I don't like this `container_id` either TBH, but I used what Datadog uses (e.g. https://github.com/DataDog/datadog-agent/blob/ff3eefad6f31c6721aac938cc25a8e6a5e5ea45a/pkg/security/telemetry/telemetry.go#L58). However, I don't see that as a strong argument to go against the convention, so I'm changing this to `container.id` as per your suggestion.